lib/bots/versionbot.ts

```diff
@@ -269,70 +334,86 @@ export class VersionBot extends GithubBot.GithubBot {
 
             // If we found a change-type message, then mark this commit as ok.
             if (changetypeFound) {
-                return this.gitCall(githubApi.repos.createStatus, {
-                    context: 'Versionist',
-                    description: 'Found a valid Versionist `Change-Type` tag',
-                    owner,
-                    repo: name,
-                    sha: head.sha,
-                    state: 'success'
+                return this.githubCall({
+                    data: {
+                        context: 'Versionist',
+                        description: 'Found a valid Versionist `Change-Type` tag',
+                        owner,
+                        repo: name,
+                        sha: head.sha,
+                        state: 'success'
+                    },
+                    method: ghApiCalls.repos.createStatus
                 });
             }
 
             // Else we mark it as having failed and we inform the user directly in the PR.
-            this.log(ProcBot.LogLevel.INFO, `${action.name}: No valid 'Change-Type' tag found, failing last commit ` +
+            this.logger.log(LogLevel.INFO, "No valid 'Change-Type' tag found, failing last commit " +
                 `for ${owner}/${name}#${pr.number}`);
-            return this.gitCall(githubApi.repos.createStatus, {
-                        context: 'Versionist',
-                        description: 'None of the commits in the PR have a `Change-Type` tag',
-                        owner,
-                        repo: name,
-                        sha: head.sha,
-                        state: 'failure'
+            return this.githubCall({
+                data: {
+                    context: 'Versionist',
+                    description: 'None of the commits in the PR have a `Change-Type` tag',
+                    owner,
+                    repo: name,
+                    sha: head.sha,
+                    state: 'failure'
+                },
+                method: ghApiCalls.repos.createStatus,
             }).then(() => {
                 // We only complain on opening. Either at that point it'll be fixed, or we
                 // simply don't pass it in future. This stops the PR being spammed on status changes.
-                if (data.action === 'opened') {
-                    this.gitCall(githubApi.issues.createComment, {
-                        body: `@${data.sender.login}, please ensure that at least one commit contains a` +
-                            '`Change-Type:` tag.',
-                        owner,
-                        number: pr.number,
-                        repo: name,
+                if (event.cookedEvent.data.action === 'opened') {
```

Comment:
Would it not be better to just warn the user of not having a change-type footer when it comes to merge time? My thinking is that PRs often have a lot of conversation going on, and it would be better to have the warnings (essentially PR metadata) occur after the main bulk of discussion, after approvals etc. Not a big sticking point for me but is probably worth thinking about.